aspnet/fundamentals/owin.rst

```diff
@@ -55,58 +56,53 @@ You can of course configure other actions to take place within the OWIN pipeline
 -----------------------------------------------------------------------
 
-Another example of how OWIN-based servers' features can be leveraged by ASP.NET Core is access to features like WebSockets. The .NET OWIN web server used in the previous example has support for Web Sockets built in, which can be leveraged by an ASP.NET Core application. The example below shows a simple web application that supports Web Sockets and echos back anything sent to the server via WebSockets.
+Another example of how OWIN-based servers' features can be leveraged by ASP.NET Core is access to features like WebSockets. The .NET OWIN web server used in the previous example has support for Web Sockets built in, which can be leveraged by an ASP.NET Core application. The example below shows a simple web app that supports Web Sockets and echoes back everything sent to the server through WebSockets.
 
 .. literalinclude:: owin/sample/src/NowinWebSockets/Startup.cs
   :lines: 11-
   :language: c#
   :linenos:
   :emphasize-lines: 7, 9-10
 
-This `sample  <https://github.com/aspnet/Docs/tree/master/aspnet/fundamentals/owin/sample>`__ is configured using the same ``NowinServer`` as the previous one - the only difference is in how the application is configured in its ``Configure`` method. A test using `a simple websocket client <https://chrome.google.com/webstore/detail/simple-websocket-client/pfdhoblngboilpfeibdedpjgfnlcodoo?hl=en>`_ demonstrates that the application works as expected:
+This `sample  <https://github.com/aspnet/Docs/tree/master/aspnet/fundamentals/owin/sample>`__ is configured using the same ``NowinServer`` as the previous one - the only difference is in how the application is configured in its ``Configure`` method. A test using `a simple websocket client <https://chrome.google.com/webstore/detail/simple-websocket-client/pfdhoblngboilpfeibdedpjgfnlcodoo?hl=en>`_ demonstrates  the application:
 
 .. image:: owin/_static/websocket-test.png
 
 
 OWIN keys
 ---------
 
-OWIN depends heavily on an ``IDictionary<string,object>`` used to communicate information throughout an HTTP Request/Response exchange. ASP.NET Core implements all of the required and optional keys outlined in the OWIN specification, as well as some of its own. Note that any keys not required in the OWIN specification are optional and may only be used in some scenarios. When working with OWIN keys, it's a good idea to review the list of `OWIN Key Guidelines and Common Keys <http://owin.org/spec/spec/CommonKeys.html>`_
+OWIN depends on an ``IDictionary<string,object>`` object to communicate information throughout an HTTP Request/Response exchange. ASP.NET Core implements all of the keys outlined in the OWIN specification, and some ASP.NET Core OWIN keys. Any keys not required in the OWIN specification are optional and may only be used in some scenarios. See `OWIN Key Guidelines and Common Keys <http://owin.org/spec/spec/CommonKeys.html>`_
```

Comment:
>Any keys not required in the OWIN specification are optional and may only be used in some scenarios.

Can you write that less nebulously?  That's totally useless. What's the limitation? Portability?